x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -384,47 +392,102 @@ const buildEqlDataProviderOrFilter = (
   return { filters: [], dataProviders: [] };
 };
 
-const createThresholdTimeline = (
+const createThresholdTimeline = async (
   ecsData: Ecs,
   createTimeline: ({ from, timeline, to }: CreateTimelineProps) => void,
   noteContent: string,
   templateValues: { filters?: Filter[]; query?: string; dataProviders?: DataProvider[] }
 ) => {
-  const { thresholdFrom, thresholdTo, dataProviders } = getThresholdAggregationData(ecsData);
-  const params = getField(ecsData, ALERT_RULE_PARAMETERS);
-  const filters = getFiltersFromRule(params.filters ?? ecsData.signal?.rule?.filters) ?? [];
-  const language = params.language ?? ecsData.signal?.rule?.language ?? 'kuery';
-  const query = params.query ?? ecsData.signal?.rule?.query ?? '';
-  const indexNames = params.index ?? ecsData.signal?.rule?.index ?? [];
-
-  return createTimeline({
-    from: thresholdFrom,
-    notes: null,
-    timeline: {
-      ...timelineDefaults,
-      description: `_id: ${ecsData._id}`,
-      filters: templateValues.filters ?? filters,
-      dataProviders: templateValues.dataProviders ?? dataProviders,
-      id: TimelineId.active,
-      indexNames,
-      dateRange: {
-        start: thresholdFrom,
-        end: thresholdTo,
-      },
-      eventType: 'all',
-      kqlQuery: {
-        filterQuery: {
-          kuery: {
-            kind: language,
-            expression: templateValues.query ?? query,
+  try {
+    const alertResponse = await KibanaServices.get().http.fetch<
+      estypes.SearchResponse<{ '@timestamp': string; [key: string]: unknown }>
+    >(DETECTION_ENGINE_QUERY_SIGNALS_URL, {
+      method: 'POST',
+      body: JSON.stringify(buildAlertsQuery([ecsData._id] ?? [])),
```

Comment:
is the nullish operator here necessary? The empty array will never be selected

Reply:
ya it's not, will remove.